File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        h = dict()
        n = 0
        for i in nums:
            h[i] = n
            n += 1
        k = 0
        for i in nums:
            res = h.get(target - i)
            if res is not None and res != k:
                return [min(k, res), max(k, res)]
            k += 1

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.ts
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers: number[], target: number): number[] {
        let a: number = 0;
        let b: number = numbers.length - 1;
        while (a < b){
            if (numbers[a] + numbers[b] < target){
                ++a
            }else if (numbers[a] + numbers[b] > target){
                --b
            }else{
                return [a + 1, b + 1];
            }
        }
        return [a,b];
    }
}
